nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py

```diff
@@ -0,0 +1,255 @@
+            return self.right_arm.wait_for_result()
+        return True
+
+    def cmd_left_arm(self, joint_values, blocking=True):
+        goal = joint_goal(joint_values, self.left_arm_joint_names)
+        self.left_arm.send_goal(goal)
+        if blocking:
+            return self.left_arm.wait_for_result()
+        return True
+
+    def home(self):
+        self.cmd_left_arm(self.left_arm_home_joint_values, blocking=False)
+        self.left_arm.wait_for_result()
+        self.cmd_right_arm(self.right_arm_home_joint_values, blocking=False)
+        self.right_arm.wait_for_result()
+        return True
+
+
+class ManipulationGripperControl:
+    def __init__(self):
+        self.left_gripper = actionlib.simple_action_client.SimpleActionClient(
+            "/moveit_action_handlers/left_arm/gripper_ctrl", MoveToJointsMoveItAction)
+        self.left_gripper.wait_for_server()
+
+        self.right_gripper = actionlib.simple_action_client.SimpleActionClient(
+            "/moveit_action_handlers/right_arm/gripper_ctrl", MoveToJointsMoveItAction)
+        self.right_gripper.wait_for_server()
+
+        self.right_open_goal = joint_goal([0.23, 0.23, 0.23], right_gripper_joint_names)
+        self.right_closed_goal = joint_goal([0, 0, 0], right_gripper_joint_names)
+
+        self.left_open_goal = joint_goal([0.23, 0.23, 0.23], left_gripper_joint_names)
+        self.left_closed_goal = joint_goal([0, 0, 0], left_gripper_joint_names)
+
+        self.gripper_lock = [0, 0]
+        self.unlock_left_gripper()
+        self.unlock_right_gripper()
+
+    def cmd_right_gripper(self, goal, blocking=True):
+        if self.gripper_lock[0]:
+            self.right_gripper.send_goal(goal)
+            if blocking:
+                return self.right_gripper.wait_for_result()
+            return True
+        return False
+
+    def cmd_left_gripper(self, goal, blocking=True):
+        if self.gripper_lock[1]:
+            self.left_gripper.send_goal(goal)
+            if blocking:
+                return self.left_gripper.wait_for_result()
+            return True
+        return False
+
+    def lock_right_gripper(self):
+        self.gripper_lock[0] = 0
+
+    def unlock_right_gripper(self):
+        self.gripper_lock[0] = 1
+
+    def lock_left_gripper(self):
+        self.gripper_lock[1] = 0
+
+    def unlock_left_gripper(self):
+        self.gripper_lock[1] = 1
+
+
+class ManipulationCartesianControl:
+    def __init__(self):
+        self.right_arm = actionlib.simple_action_client.SimpleActionClient(
+            "/moveit_action_handlers/right_arm/cartesian_ctrl", MoveToPoseMoveItAction)
+        self.right_arm.wait_for_server()
+
+        self.left_arm = actionlib.simple_action_client.SimpleActionClient(
+            "/moveit_action_handlers/left_arm/cartesian_ctrl", MoveToPoseMoveItAction)
+        self.left_arm.wait_for_server()
+        self.ee_ctrl_mode = 0
+
+    def cmd_right_arm(self, pose: Pose, blocking=True):
+        pass
```

Comment:
nit: raise not implemented. this is a silent failure that might be tricky if we mess up elsewhere